src/cmd/script.rs

```diff
@@ -92,10 +94,10 @@ pub fn run(sess: &Session, matches: &ArgMatches) -> Result<()> {
 /// Files with the same category that appear after each other will be kept in
 /// the same source group. Files with different cateogries are split into
 /// separate groups.
-fn separate_files_in_group<F1, F2, T>(mut src: SourceGroup, categorize: F1, consume: F2)
+fn separate_files_in_group<F1, F2, T>(mut src: SourceGroup, categorize: F1, mut consume: F2)
 where
     F1: Fn(&SourceFile) -> Option<T>,
-    F2: Fn(&SourceGroup, T, Vec<SourceFile>),
+    F2: FnMut(&SourceGroup, T, Vec<SourceFile>),
```

Comment:
@fabianschuiki: I had to change this and line 97 to be able to manipulate `defines` from inside the closure on line 333. Was this the proper way to do so?

Reply:
Yes this is perfect!